File: uart/RaspberryPi2/wiringPi/UART.c
#include <stdio.h>
#include <wiringPi.h>
#include <wiringSerial.h>

int main()
{
	int fd;
	if(wiringPiSetup() < 0)return 1;
	if((fd = serialOpen("/dev/ttyAMA0",115200)) < 0)return 1;
//	if((fd = serialOpen("/dev/ttyS0",115200)) < 0)return 1;
	printf("serial test start ...\n");
	serialPrintf(fd,"Hello World!!!\n");
	while(1)
	{
		serialPutchar(fd,serialGetchar(fd));
	}
	serialClose(fd);
	return 0;
}

File: uart/RaspberryPi3/python/uart.py
#!/usr/bin/python
# -*- coding:utf-8 -*-
import serial

#ser = serial.Serial("/dev/ttyAMA0",115200)
ser = serial.Serial("/dev/ttyS0",115200)

print('serial test begin')
ser.write("Hello World !!!\n")
try:
	while True:
		ser.write(ser.read())
except KeyboardInterrupt:
	if ser != None:
		ser.close()
